Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -189,31 +182,29 @@ private func createResolvedPackages(
     allManifests: [GraphLoadingNode],
     mirrors: DependencyMirrors,
     manifestToPackage: [Manifest: Package],
-    // FIXME: This shouldn't be needed once <rdar://problem/33693433> is fixed.
-    rootManifestSet: Set<Manifest>,
     unsafeAllowedPackages: Set<PackageReference>,
     diagnostics: DiagnosticsEngine
 ) -> [ResolvedPackage] {
-
     // Create package builder objects from the input manifests.
-    let packageBuilders: [ResolvedPackageBuilder] = allManifests.compactMap({ node in
+    let packageBuilders: [ResolvedPackageBuilder] = allManifests.compactMap { node in
         guard let package = manifestToPackage[node.manifest] else {
             return nil
         }
-        let isAllowedToVendUnsafeProducts = unsafeAllowedPackages.contains{ $0.path == package.manifest.url }
+        let isAllowedToVendUnsafeProducts = unsafeAllowedPackages.contains { $0.path == package.manifest.url }
         return ResolvedPackageBuilder(
             package,
             productFilter: node.productFilter,
             isAllowedToVendUnsafeProducts: isAllowedToVendUnsafeProducts
         )
-    })
+    }
 
     // Create a map of package builders keyed by the package identity.
-    let packageMapByIdentity: [String: ResolvedPackageBuilder] = packageBuilders.spm_createDictionary{
-        let identity = PackageReference.computeIdentity(packageURL: $0.package.manifest.url)
-        return (identity, $0)
-    }
-    let packageMapByName: [String: ResolvedPackageBuilder] = packageBuilders.spm_createDictionary{ ($0.package.name, $0) }
+    let packageMapByIdentity: [String: ResolvedPackageBuilder] = Dictionary(uniqueKeysWithValues:
+        packageBuilders.map { (PackageIdentity($0.package.manifest.url).computedName, $0) }
+    )
+    let packageMapByName: [String: ResolvedPackageBuilder] = Dictionary(uniqueKeysWithValues:
+        packageBuilders.map { ($0.package.name, $0) }
+    )
```

Comment:
lookup by string instead of PackageIdentity intentional?

Reply:
Looks like it, since two different mappings are being built up.

Reply:
@mattt are we using the legacy identity here because this is the part we are careful no to change semantically?

Reply:
@tomerd @abertelrud Correct. This diff makes no changes to semantics or behavior.